pkg/kubernetes/portforward.go

```diff
@@ -0,0 +1,139 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package kubernetes
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	core_v1 "k8s.io/api/core/v1"
+	k8s "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/portforward"
+	"k8s.io/client-go/transport/spdy"
+)
+
+type PortForward struct {
+	Config     *rest.Config
+	Method     string
+	URL        *url.URL
+	Host       string
+	LocalPort  int
+	RemotePort int
+	EmitLogs   bool
+	StopCh     chan struct{}
+	ReadyCh    chan struct{}
+}
+
+func NewPortForward(
+	config *rest.Config,
+	namespace, deployName string,
+	host string, localPort, remotePort int,
+	emitLogs bool,
+) (*PortForward, error) {
+	time.Sleep(10 * time.Second)
```

Comment:
I think this sleep is not needed here, almost it is quite long 10 secs.

Reply:
Agree, 10 secs feel a bit long. I had to introduce this 10 secs delay because portforwarding would fail since the pod takes couple of seconds to be up and running. Any ideas how to do this the right way using client API?

Reply:
Just looked into it, does this help?

https://gianarb.it/blog/programmatically-kube-port-forward-in-go

Reply:
Yes, that's the way to do it. Sleep is not needed.

Generally, we should only connect if the dashboard pod is in Ready status.